python/cudf/cudf/core/indexed_frame.py

```diff
@@ -2040,9 +2034,12 @@ def interpolate(
                 col = col.astype("float64").fillna(np.nan)
 
             # Interpolation methods may or may not need the index
-            columns[colname] = interpolator(col, index=data.index)
+            columns.append(interpolator(col, index=data.index))
 
-        result = self._from_data(columns, index=data.index)
+        result = self._from_data_like_self(
+            self._data._from_columns_like_self(columns)
+        )
+        result.index = data.index
```

Comment:
I'm not sure historically why `_from_data_like_self` doesn't take an index argument. I'm hoping by definition this _wouldn't_ take an index argument so that the "like self" means "preserve the index"

Reply:
thanks

Reply:
Yes, that is exactly the intention. See https://github.com/rapidsai/cudf/pull/11140
> Specifically in instances where the outputs need to share the index of the input for IndexedFrame methods, this hook provides the necessary entrypoint